receiver/jaegerreceiver/config.go

```diff
@@ -30,12 +30,19 @@ type RemoteSamplingConfig struct {
 	configgrpc.GRPCClientSettings `mapstructure:",squash"`
 }
 
+type SecureSetting struct {
+	configmodels.ReceiverSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+	// Configures the receiver to use TLS.
+	// The default value is nil, which will cause the receiver to not use TLS.
+	TLSCredentials *configtls.TLSSetting `mapstructure:"tls_credentials, omitempty"`
```

Comment:
`tls_credentials` makes sense for a server scenario but for a client they must just be be configuring the CA but no client certificates so there's no credentials. In those cases I think we'd want to name this `tls_config` or just `tls`. Would it make sense to have a universal config key for the TLS settings and call this the same thing here?

Reply:
Yeah renaming the config field is an option and something that should be figured out in a separate issue. The goal of this pr is to deprecate the SecureReceiverSettings used by several server connection scenarios.